### PR TITLE
ADL-S support to SOF and legacy driver

### PR DESCRIPTION
**Pull request by kv2019i**

### CODE REVIEW

File: sound/soc/sof/sof-pci-dev.c [hunk 1]
```diff
@@ -284,6 +284,23 @@ static const struct sof_dev_desc jsl_desc = {
 };
 #endif
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_ALDERLAKE)
+static const struct sof_dev_desc adls_desc = {
+	.machines		= snd_soc_acpi_intel_hda_machines,
```
Comment (kv2019i) [inline sound/soc/sof/sof-pci-dev.c:289]: I did not add any dummy acpi tables yet, but instead pointed machines to hda_machines for now. Later patches will add the ACPI machine tables if/when needed.

File: sound/soc/sof/sof-pci-dev.c [hunk 2]
```diff
@@ -284,6 +284,23 @@ static const struct sof_dev_desc jsl_desc = {
...
+	.resindex_lpe_base      = 0,
+	.resindex_pcicfg_base   = -1,
+	.resindex_imr_base      = -1,
+	.irqindex_host_ipc      = -1,
+	.resindex_dma_base      = -1,
+	.chip_info = &adls_chip_info,
+	.default_fw_path = "intel/sof",
+	.default_tplg_path = "intel/sof-tplg",
+	.default_fw_filename = "sof-adl-s.ri",
```
Comment (kv2019i) [inline sound/soc/sof/sof-pci-dev.c:298]: Note: this is sof-adl-s.ri now (not sof-adls.ri -> FYI especially to @xiulipan ).

**Review by plbossart — APPROVED** [review]

File: sound/soc/sof/sof-pci-dev.c (hunk 2 above)
Comment (xiulipan) [reply sound/soc/sof/sof-pci-dev.c:298]: @kv2019i Noted, we will make this as our release and CI binary naming.

File: sound/soc/sof/sof-pci-dev.c [hunk 3]
```diff
@@ -284,6 +284,23 @@ static const struct sof_dev_desc jsl_desc = {
...
+	.nocodec_tplg_filename = "sof-adl-nocodec.tplg",
```
Comment (xiulipan) [inline sound/soc/sof/sof-pci-dev.c:299]: do we also need to use `sof-adl-s-nocodec.tplg` here? If there is any different clock setting and SSP configuration with different ADL platforms.

Comment (kv2019i) [reply sound/soc/sof/sof-pci-dev.c:299]: @xiulipan I considered this, but for now we have no additional needs to variate, so I propose to stick to similar approach than we had with TGL and just have one nocodec file for the family.

**Review by lyakh — APPROVED** [review]

**Pull request merged**